File: Remove Element/sol2.cpp

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        //two pointers
            int i = 0;//slow pointer, and the length of the new array
            for (int j = 0; j < nums.size(); j++) {
                if (nums[j] != val) {//we write selectively, if num[j]==val, we need to remove by overwriting the pos
                    nums[i] = nums[j];
                    i++;
                }
            }
            return i;
    }
};

File: Container With Most Water/sol2.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
	   #Two pointers O(n)
        L, R = 0, len(height)-1
        max_area = 0 
        while L<R:
            length = R- L
            if height[L] < height[R]:
                h = height[L]
                L+=1
            elif height[L] > height[R]:
                h = height[R]
                R-=1
            else:
                h = height[L]
                L+=1
      
            max_area = max(max_area, h*length)
        
        return max_area